x-pack/plugins/observability_solution/apm/public/components/app/service_tab_content/types.ts

```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export type Tab = 'overview' | 'dependencies' | 'infrastructure' | 'service-map';
```

Comment:
Can use use the Tab['key'] that are already defined [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/public/components/routing/templates/apm_service_template/index.tsx#L54)

Reply:
Yes, that could be used, but the idea was to build a list only for the tabs that are enabled to work with ServiceTabContent and not necessarily tie it to the routing tabs, as I see them as somewhat different domains. What do you think?

Reply:
So, in this case, I think we're mixing up contexts here. You created a new HOC called <ServiceTabContent .../> which expects a `tabName` property and wrapped some pages with it. The problem I see here is that the Page components (service maps, service dependencies, infrastructure...) don't know if they will be rendered on a tab or not. What controls it is the **ApmServiceTemplate**. 

I think we should either link this `Tab` to the `Tab` defined in the `ApmServiceTemplate` or rename the HOC and its types to something else other than tab. WDYT?

Reply:
Oh, I see! I thought those components were meant exclusively for use as tabs, but it’s true that there are no constraints on that.

Let’s keep it simple and use the routing tabs here, as it seems like the more straightforward approach for now. Although, I do like the idea of renaming the component and rethinking its usage.

Thank you so much for the detailed suggestion!